src/uu/fmt/src/fmt.rs

```diff
@@ -141,21 +137,40 @@ fn parse_arguments(args: impl uucore::Args) -> UResult<(Vec<String>, FmtOptions)
                 ));
             }
         };
-        if fmt_opts.width > MAX_WIDTH {
-            return Err(USimpleError::new(
-                1,
-                format!(
-                    "invalid width: '{}': Numerical result out of range",
-                    fmt_opts.width,
-                ),
-            ));
-        }
-        fmt_opts.goal = cmp::min(
-            fmt_opts.width * DEFAULT_GOAL_TO_WIDTH_RATIO / 100,
-            fmt_opts.width - 3,
-        );
     };
 
+    match matches.get_one::<String>(OPT_GOAL) {
+        Some(s) => {
+            fmt_opts.goal = match s.parse::<usize>() {
+                Ok(t) => t,
+                Err(e) => {
+                    return Err(USimpleError::new(
+                        1,
+                        format!("Invalid GOAL specification: {}: {}", s.quote(), e),
+                    ));
+                }
+            };
+            if !matches.contains_id(OPT_WIDTH) {
+                // GNU fmt.c:407
```

Comment:
Please don't reference the GNU source code, we shouldn't look at it (see https://github.com/uutils/coreutils/blob/main/CONTRIBUTING.md#best-practices):
```suggestion
                // https://www.gnu.org/software/coreutils/manual/html_node/fmt-invocation.html#index-_002dwidth
```